### ISSUE TITLE
Support get list of ipv4 by location in datadog_ip_ranges data source

### ISSUE THREAD

**Issue by kuanhn** [DataDog/terraform-provider-datadog#884]
### Description

Because in the Synthetics test, sometimes we just want to use certain locations, we want to be able to improve the source so that we can get the IP in the `prefixes_ipv4_by_location` field by location.

https://docs.datadoghq.com/api/latest/ip-ranges/

### New or Affected Resource(s)

* datadog_ip_ranges

### Potential Terraform Configuration

```tf
data "datadog_ip_ranges" "test" {}

resource "aws_waf_ipset" "datadog_ip" {
  name = "datadog_ip"

  dynamic "ip_set_descriptors" {
    for_each = data.datadog_ip_ranges.test.synthetics_ipv4_by_location.ap-northeast-1

    content {
      type  = "IPV4"
      value = ip_set_descriptors.value
    }
  }
}
```

**Comment by egorchabala** [comment]
Might be very convenient - we have a network firewall rule that allows connection from Datadog synthetic locations. The current process to get Datadog IPs is implying parsing JSON and using it inside Terraform code. The requested data resource will make the configuration simple and straightforward.

**Comment by fanny-jiang** [comment]
Hey @kuanhn and @egorchabala, I'm working on a [PR](https://github.com/DataDog/terraform-provider-datadog/pull/934) to add support for this feature (see above). The API returns a map of locations and list of IPs for each location, however due to limitations in the Terraform SDK, creating a map of lists isn't supported, so the above configuration wouldn't work. In my PR, the state would be set with a map of locations and IPs as a concatenated string, delimited by `,`. It's not pretty, but you would be able to do a split on the string to get the IPs by location. All other potential schemas and data structures would make the configuration very cumbersome. Let me know your thoughts.

**Comment by therve** [comment]
Closes with #934

**Comment by kuanhn** [comment]
@fanny-jiang 
It's great. That will help us a lot. Thank you.
